Prose/ProseLib/Sources/ConversationFeature/MessageBar/MessageBarTextField.swift

```diff
@@ -62,42 +69,76 @@ struct MessageBarTextField: View {
 
 // MARK: Reducer
 
+enum MessageBarTextFieldEffectToken: Hashable, CaseIterable {
+  case pauseTyping
+}
+
+public extension DispatchQueue.SchedulerTimeType.Stride {
+  static let pauseTypingDebounceDuration: Self = .seconds(30)
+}
+
 public let messageBarTextFieldReducer: Reducer<
   MessageBarTextFieldState,
   MessageBarTextFieldAction,
-  Void
-> = Reducer { state, action, _ in
+  ConversationEnvironment
+> = Reducer { state, action, environment in
   switch action {
   case .sendTapped where !state.message.isEmpty:
     let content = state.message
     state.message = ""
     return Effect(value: .send(message: content))
 
-  default:
+  case .binding(\.$isFocused) where state.isFocused && !state.message.isEmpty,
+       .binding(\.$message) where !state.message.isEmpty:
+    return .merge([
```

Comment:
I always use `.concatenate`, but this time I told myself that we don't need the two effects to run one after another, so I used `.merge`.

Should we go for `.merge` by default unless we need effects to run in a certain order or go for `.concatenate` by default unless we need effects to run in *parallel(?)*?

Reply:
The former. Run in parallel unless there's a good reason (a dependency most likely) not to.